be/src/runtime/result_queue_mgr.cpp

```diff
@@ -79,7 +79,12 @@ Status ResultQueueMgr::fetch_result(const TUniqueId& fragment_instance_id, std::
     } else {
         *eos = true;
     }
-    return Status::OK();
+    if (UNLIKELY(*result == nullptr)) {
+        // return the status of the queue, in case the status is updated during the blocking_get()
```

Comment:
Why add this judgment?

Reply:
there are two cases that the result can be `nullptr`, 1. Sunny case: the fragment execution done, no more data there in the queue. 2. raining case: execution failed, the fragment is canceled, nullptr is added into the result queue in order to wake up all the waiters.

In both cases, the result will be nullptr, but the queue's status will be different.